### DISCUSSION TITLE
Memory consumption increased with version from 25. Sep

### DISCUSSION THREAD

**Discussion by adaumann** [drmortalwombat/oscar64#154] [General]
Hi,

I downloaded the latest OscarC from 25.9. I'm using the -Os flag to keep the read-write data and the C library in main memory as small as possible. The rest is stored in easyflash bank. With the new version I noticed that the compiler produces larger code or the libraries are larger? Therefore my project is no longer building. Is there a way to get the smaller size again?
What does: "Main region does not fit in first ROM bank?" mean?
Currently I stick to an older version (1 month ago)

andy

**Comment by drmortalwombat** [comment]
Without your source or more detail, it is hard for me to guess what the issue might be.  There have not been any significant increases in the libraries.  There are some changes in the compiler to get it more compliant with C (e.g. constant initializer of local variables).

**Reply by adaumann** [reply]
Ok, I don't like to share my source code in GitHub right now (perhaps later after release). Could you give me an email? But be warned, it will be not a very clean code in C like manner...

**Reply by drmortalwombat** [reply]
You can send me a github invite

**Reply by adaumann** [reply]
Ok done, I moved to GitHub and send an invite...

**Reply by drmortalwombat** [reply]
Thank you, this is very helpful.  I keep you posted on the progress.  If you want to chat you can usually find me on the EverythingC64 discord during CET daylight hours.

**Reply by drmortalwombat** [reply]
Your build.bat uses command line option -nz which is not valid, I assume you want -xz -Oz .  I will add more error checking on the command line options with the next release.

**Reply by adaumann** [reply]
Hi,

if I choose following flags: "oscar64 .\orbix.cpp  -n -g -Os -tf=crt -xz -Oz", the game fits again into memory but crashes during init of the first level. I assume I need to assign variables manually to zeropages instead of automated...

**Reply by drmortalwombat** [reply]
proxy0_spr_set  needs no inline or it gets inlined into rom at some points, causing the crash.

**Reply by drmortalwombat** [reply]
You can also save some memory by not using the heap, there is only one malloc in the code, and that could be easily a local variable.

**Reply by adaumann** [reply]
I added a __noinline to the proxy0_spr_set: __noinline void proxy0_spr_set(char sp, bool show, int xpos, int ypos, char image, char color, bool multi, bool xexpand, bool yexpand)

But no success... what exactly you mean?

**Reply by drmortalwombat** [reply]
New version/installer is up, that should help with your problems

**Reply by adaumann** [reply]
I really appreciate how quickly you solve issues, now it works again. All flag combinations are working, -xz, -Oz and -Os only. 

 With -xz flag enabled there is a different behaviour when the ball hits the bomb. Without flags the elements in radius are deleted, with flag there are many more deleted. But for me it works with zero flags
 
 Many thanks,
 andy

**Comment by adaumann** [comment]
BTW: I noticed that some arguments like -O or not accepted anymore... seems that there is something wrong

**Reply by drmortalwombat** [reply]
Fix checked in.  -O and -O1 are default, so just pass no -O option as a workaround until a new release is uploaded, if you don't build oscar yourself.
